docs/src/user/algorithms.rst

```diff
@@ -186,6 +186,71 @@ converging trials that do not lead to best results at convergence (stragglers).
 To overcome this, you can increase the number of brackets, which increases the amount of resources
 required for optimisation but decreases the bias towards stragglers. Default is 1.
 
+
+.. _tpe-algorithm:
+
+TPE
+---------
+
+`Tree-structured Parzen Estimator`_ (TPE) algorithm is one of Sequential Model-Based
+Global Optimization (SMBO) algorithms, which will build models to propose new points based
+on the historical observed trials.
+
+Instead of modeling p(y|x) like other SMBO algorithms, TPE models p(x|y) and p(y),
+and p(x|y) is modeled by transforming that generative process, replacing the distributions of
+the configuration prior with non-parametric densities.
+
+The TPE defines p(x|y) using two such densities l(x) and g(x) while l(x) is distribution of
+good points and g(x) is the distribution of bad points. New point candidates will be sampled
```

Comment:
This sentence or a next sentence should state how good and bad points are defined by a quantile `\gamma` so that a point for which y < y* is a good one and y >= y* is a bad one, where y* is defined based on \gamma = p(y < y*).

Reply:
Maybe: Good and bad points are defined as the `gamma` best points and `1 - gamma` worst points observed so far.